modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorMatchedSlotSubFetchPhase.java

```diff
@@ -56,62 +56,86 @@
     static final String FIELD_NAME_PREFIX = "_percolator_document_slot";
 
     @Override
-    public void hitsExecute(SearchContext context, SearchHit[] hits) throws IOException {
-        innerHitsExecute(context.query(), context.searcher(), hits);
-    }
+    public FetchSubPhaseProcessor getProcessor(SearchContext searchContext) throws IOException {
 
-    static void innerHitsExecute(Query mainQuery,
-                                 IndexSearcher indexSearcher,
-                                 SearchHit[] hits) throws IOException {
-        List<PercolateQuery> percolateQueries = locatePercolatorQuery(mainQuery);
-        if (percolateQueries.isEmpty()) {
-            return;
+        List<PercolateContext> percolateContexts = new ArrayList<>();
+        for (PercolateQuery pq : locatePercolatorQuery(searchContext.query())) {
+            percolateContexts.add(new PercolateContext(pq));
+        }
+        if (percolateContexts.isEmpty()) {
+            return null;
         }
+        boolean singlePercolateQuery = percolateContexts.size() == 1;
```

Comment:
Small comment, we could pass this into the `PercolateContext` constructor to avoid always having to pass it into `PercolateContext#fieldName`.